### ISSUE TITLE
Curver arrow adding a white space between head and line

### ISSUE THREAD

**Issue by davibarreira** [JuliaGraphics/Luxor.jl#200]
I'm trying to create a self-referential loop using `arrow`, but for some reason, it's adding a white space between the
line and the arrow head. Any ideas on why that is? From the examples in the docs, I didn't figure out why this is happening.
```julia
d = Drawing(100,100,:svg)
origin()
circle(O, 5,:fill)
fontsize(12)
label(L"a",:S, offset=10)
arrow(Point(-8,-2),Point(-50,-50), Point(50,-50),Point(8,-2),arrowheadlength = 10)
finish()
d
```
![image](https://user-images.githubusercontent.com/6407557/154947519-f1a27154-a4a5-485d-8c7c-2d45cc2b0bf5.png)

**Comment by cormullion** [comment]
As for "why", I'm  guessing that I didn't examine all the possibilities of adding Bezier arrows... 🤣 The arrows code got very complicated for no really good reason, and there are just so many possibilities, not all of them are fully tested. It could probably all do with rewriting one day... 

I suppose measuring distances along arcs and Bezier curves requires separate code paths.
 
You might  be able to work around this bug by drawing two arrows:

```julia
arrow(Point(-8,-2),
	Point(-50,-50), 
	Point(50,-50),
	Point(8,-2),
	arrowheadlength = 10,
	finisharrow=false
	)
arrow(Point(-8,-2),
	Point(-50,-50), 
	Point(50,-50),
	Point(8,-2),
	linewidth = 0,
	arrowheadlength = 10,
	finisharrow=true
	)
```

<img width="102" alt="Screenshot 2022-02-21 at 13 07 00" src="https://user-images.githubusercontent.com/52289/154961197-d86b92f7-c213-41d9-99c6-35cbbc5568d8.png">
 
but I know that's a bit hacky :)

**Comment by davibarreira** [comment]
Very nice! I ended up writing another arrowheadfunction.
```julia
function quiverarrow(shaftendpoint, endpoint, shaftangle)
    @layer begin
        r = 6
        setlinecap("round")
        translate(endpoint)
        rotate(shaftangle-π)
        # setline(1)
        
        c1  = Point(0,-r)
        p1 = c1 + r*Point(cos(π/10), sin(π/10))
        p2 = O
        
        arc2r(c1,p1,p2,:stroke)

        c2  = Point(0,r)
        p3 = c2 + r*Point(cos(-π/10), sin(-π/10))
        arc2r(c2,p2,p3,:stroke)
    end
end


d = Drawing(100,100,:svg)
origin()
circle(O, 5,:fill)
fontsize(12)
label(L"a",:S, offset=10)
arrow(Point(-8,-2),Point(-50,-50), Point(50,-50),Point(8,-2),linewidth=1,
    arrowheadlength = 0,
    arrowheadfunction = quiverarrow)

arrow(O, Point(50,0),
    arrowheadlength = 0,
    arrowheadfunction = quiverarrow)
finish()
d
```
![image](https://user-images.githubusercontent.com/6407557/154962247-b1cc4c2c-15b3-416d-9da4-890cc0b57f80.png)

**Comment by cormullion** [comment]
That's great! Hopefully the bug will  be fixed one day... :)

**Comment by cormullion** [comment]
Might be fixed for next release  :) 

<img width="548" alt="Screenshot 2022-02-22 at 13 17 31" src="https://user-images.githubusercontent.com/52289/155140290-9b272c5b-3cb5-4e49-885f-322a24989d84.png">

**Comment by hyrodium** [comment]
The original issue was fixed with release `v3.1.0` ([CHANGELOG.md](https://github.com/JuliaGraphics/Luxor.jl/blob/master/CHANGELOG.md#v310---2022-02-26))

```julia
using Luxor
using Colors

d = Drawing(100,100,joinpath(@__DIR__, "a.svg"))
origin()
setcolor(RGBA(1,0,0,0.5))
circle(O, 5,:fill)
fontsize(12)
arrow(Point(-8,-2),Point(-50,-50), Point(50,-50),Point(8,-2),arrowheadlength = 10)
finish()
d
```

![a](https://user-images.githubusercontent.com/7488140/173232987-d8e9ee20-3f3f-4c17-ab7d-79806df1ad57.svg)

![image](https://user-images.githubusercontent.com/7488140/173233014-d41e0b0e-bb88-49c2-9ad7-324ddc29dccd.png)

But the color issue with the overlapping translucent arrows seems to remain. (I understand this would be hard to fix..)

**Comment by cormullion** [comment]
Yes. 😀 I was just using transparency to show the overlap - I hope people don’t use it with arrows 😂😂😂
